src/simulators/statevector/statevector_executor.hpp

```diff
@@ -1260,9 +1260,17 @@ std::vector<reg_t> Executor<state_t>::sample_measure(const reg_t &qubits,
 template <class state_t>
 void Executor<state_t>::apply_initialize(const reg_t &qubits,
                                          const cvector_t &params,
-                                         RngEngine &rng) {
+                                         RngEngine &rng, bool recursive) {
```

Comment:
I think we do not need to change this method signature:
```
template <class state_t>
void Executor<state_t>::apply_initialize(const reg_t &qubits,
                                         const cvector_t &params_,
                                         RngEngine &rng) {
  auto sorted_qubits = qubits;
  std::sort(sorted_qubits.begin(), sorted_qubits.end());

  cvector_t tmp;
  // apply global phase here
  if (Base::states_[0].has_global_phase()) {
    tmp.resize(params.size());
    for (int_t i = 0; i < params.size(); i++) {
      tmp[i] = params[i] * Base::states_[0].global_phase();
    }
  }
  const cvector_t& params = tmp.empty() ? params_: tmp;
```

`omp parallel` maybe necessary if `param` exceeds a threshold.